### PR TITLE
fix: show blocked reservations as closed time slots

### PR DESCRIPTION
**Pull request by vincit-matu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Blocked reservations should show as the reservation unit being closed, so adds a check for `isBlocked` and applies same styling as for when the reservation unit is closed.
- Assumes that we also want to hide the time (and/or other possible text) so sets `color` to `transparent`.

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Create an event/a reservation on the admin side which should be seen as blocked on the client side
- Check the reservation unit's page: the reservation should show as closed
- Same thing for EditStep0 (== edit reservation, first step)

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- TILA-3016

### CODE REVIEW

File: apps/ui/components/reservation/EditStep0.tsx [hunk 1]
```diff
@@ -44,6 +38,7 @@ import { BlackButton, MediumButton } from "@/styles/util";
 import Legend from "../calendar/Legend";
 import ReservationCalendarControls from "../calendar/ReservationCalendarControls";
 import { CalendarWrapper } from "../reservation-unit/ReservationUnitStyles";
+import { eventStyleGetter } from "@/pages/reservation-unit/[id]";
```
Comment (joonatank) [inline apps/ui/components/reservation/EditStep0.tsx:41]: I'd rather not import from @pages. Could we move it to new or something `components/reservation` or `components/reservationCommon`. Something temporary is fine.

**Review by joonatank — COMMENTED** [review]
Great, getting rid of the extra `eventStyleGetter` function, but would change the export location.

File: apps/ui/components/reservation/EditStep0.tsx (hunk 1 above)
Comment (vincit-matu) [reply apps/ui/components/reservation/EditStep0.tsx]: Made a new `calendarUtils.ts` file under `components/common`.

**Review by joonatank — APPROVED** [review]
👍

**Pull request merged**